src/kontrol/foundry.py

```diff
@@ -1229,6 +1292,31 @@ def print_node(self, kcfg: KCFG, node: KCFG.Node) -> list[str]:
             if not self.omit_unstable_output and srcmap_data is not None:
                 path, start, end = srcmap_data
                 ret_strs.append(f'src: {str(path)}:{start}:{end}')
+
+        calldata_cell = node.cterm.try_cell('CALLDATA_CELL')
+        program_cell = node.cterm.try_cell('PROGRAM_CELL')
+
+        if type(program_cell) is KToken:
+            selector_bytes = None
+            if type(calldata_cell) is KToken:
+                selector_bytes = ast.literal_eval(calldata_cell.token)
+                selector_bytes = selector_bytes[:32]
+            elif (
+                type(calldata_cell) is KApply and calldata_cell.label.name == '_+Bytes__BYTES-HOOKED_Bytes_Bytes_Bytes'
+            ):
+                first_bytes = flatten_label(label='_+Bytes__BYTES-HOOKED_Bytes_Bytes_Bytes', kast=calldata_cell)[0]
+                if type(first_bytes) is KToken:
+                    selector_bytes = ast.literal_eval(first_bytes.token)
```

Comment:
I think that we need the first four bytes only? It is possible for the first element of symbolic calldata`+Bytes` to be longer than 4 bytes, think of a function where the first parameter passed in is concrete and the second id symbolic.

Reply:
Fixed.